solidity/contracts/converter/ConverterUpgrader.sol

```diff
@@ -246,24 +246,16 @@ contract ConverterUpgrader is IConverterUpgrader, ContractRegistryClient {
     function transferReserveBalancesVersion45(ILegacyConverterVersion45 _oldConverter, IConverter _newConverter)
         private
     {
-        uint256 reserveBalance;
         uint16 reserveTokenCount = _oldConverter.connectorTokenCount();
-
         for (uint16 i = 0; i < reserveTokenCount; i++) {
             IReserveToken reserveToken = _oldConverter.connectorTokens(i);
-            // Ether reserve
 
-            if (reserveToken.isNativeToken()) {
-                if (address(_oldConverter).balance > 0) {
+            uint256 reserveBalance = reserveToken.balanceOf(address(_oldConverter));
+            if (reserveBalance > 0) {
+                if (reserveToken.isNativeToken()) {
                     _oldConverter.withdrawETH(address(_newConverter));
-                }
-            }
-            // ERC20 reserve token
-            else {
-                IReserveToken connector = reserveToken;
-                reserveBalance = connector.balanceOf(address(_oldConverter));
-                if (reserveBalance > 0) {
-                    _oldConverter.withdrawTokens(connector, address(_newConverter), reserveBalance);
+                } else {
+                    _oldConverter.withdrawTokens(reserveToken, address(_newConverter), reserveBalance);

```

Comment:
Makes me wonder why we don't have a single function `withdrawReserves`.

Reply:
We do. It's called `transferReservesOnUpgrade`. These the `withdrawETH` and `withdrawTokens` are old converters' we need to support :(